src/controllers/payment.controller.ts

```diff
@@ -0,0 +1,48 @@
+import { inject } from 'inversify'
+import { controller, httpGet } from 'inversify-express-utils'
+import { RedirectResult } from 'inversify-express-utils/dts/results'
+
+import BaseController from './base.controller'
+import PaymentPostResponse from 'app/models/dto/paymentPostResponse'
+import DissolutionSession from 'app/models/session/dissolutionSession.model'
+import { PAYMENT_URI } from 'app/paths'
+import PaymentService from 'app/services/payment/payment.service'
+import SessionService from 'app/services/session/session.service'
+import TYPES from 'app/types'
+import { v4 as uuidv4 } from 'uuid'
+
+@controller(PAYMENT_URI, TYPES.SessionMiddleware, TYPES.AuthMiddleware, TYPES.CompanyAuthMiddleware)
+export class PaymentController extends BaseController {
+
+  public constructor(
+    @inject(SessionService) private session: SessionService,
+    @inject(PaymentService) private paymentService: PaymentService
+  ) {
+    super()
+  }
+
+  @httpGet('')
+  public async get(): Promise<RedirectResult> {
+    const token: string = this.session.getAccessToken(this.httpContext.request)
+    const dissolutionSession: DissolutionSession = this.session.getDissolutionSession(this.httpContext.request)!
+
+    const paymentStateUUID: string = uuidv4()
+
+    const response: PaymentPostResponse = await this.paymentService.getPaymentPage(token, dissolutionSession, paymentStateUUID)
```

Comment:
could this service just return the URL to  offload as much logic as possible away from the controller?